Tensile/Tests/unit/test_HardwarePredicates.py

```diff
@@ -82,3 +83,21 @@ def test_predicate_library_merge(libraries):
     assert lib.rows[1]['predicate'] == HardwarePredicate.FromHardware((9,0,8), 60)
     for r in lib.rows[:-1]:
         assert r['predicate'] != HardwarePredicate('TruePred')
+
+def hardware_library_objects_dups():
+    objs = [PredicateLibrary('Hardware', [{'predicate': HardwarePredicate.FromISA((9,0,0)), 'library': PredicateLibrary()}]),
+            PredicateLibrary('Hardware', [{'predicate': HardwarePredicate.FromISA((9,0,6)), 'library': PredicateLibrary()}]),
+            PredicateLibrary('Hardware', [{'predicate': HardwarePredicate.FromISA((9,0,6)), 'library': PredicateLibrary()}]),
+            PredicateLibrary('Hardware', [{'predicate': HardwarePredicate.FromISA((9,0,8)), 'library': PredicateLibrary()}]),
+            PredicateLibrary('Hardware', [{'predicate': HardwarePredicate('TruePred'),      'library': PredicateLibrary()}])
+    ]
+
+    return [copy.deepcopy(libs) for libs in itertools.permutations(objs)]
+
+@pytest.mark.parametrize("libraries", hardware_library_objects_dups())
+def test_hardware_library_merge_dups(libraries):
+    lib = libraries[0]
+    for lib2 in libraries[1:]:
+        lib.merge(lib2)
+
+    assert len(lib.rows) == 4
```

Comment:
This will probably be good enough, but it might be more reliable to test against the expected result, since it's possible for the merge  function to remove the wrong library and still pass.

Reply:
Sure. It's a pretty unlikely edge case, but testing to ensure against it should be simple enough.